torch/testing/_internal/optests/aot_autograd.py

```diff
@@ -118,7 +118,9 @@ def check(args, ignore_failure=False):
             raise
 
         # See https://github.com/pytorch/pytorch/pull/98960#issuecomment-1505962215
-        if all(x is None for x in orig_grad):
+        tensor_args = [x for x in pytree.tree_flatten(args)[0] if isinstance(x, torch.Tensor)]
+        any_non_leaves = any(x.requires_grad and not x.is_leaf for x in tensor_args)
```

Comment:
Same as above:
```suggestion
        any_non_leaves = any(x.grad_fn is not Nonefor x in tensor_args)
```

Reply:
also an interesting error from CI:
```
2024-08-22T07:36:12.9280359Z [31m[1m_ TestEagerFusionModuleInfoCPU.test_aot_autograd_module_exhaustive_nn_AdaptiveAvgPool1d_cpu_float32 _[0m
2024-08-22T07:36:12.9281197Z Traceback (most recent call last):
2024-08-22T07:36:12.9282087Z   File "/var/lib/jenkins/workspace/test/functorch/test_aotdispatch.py", line 6251, in test_aot_autograd_module_exhaustive
2024-08-22T07:36:12.9283196Z     _test_aot_autograd_module_helper(self, device, dtype, training, module_info)
2024-08-22T07:36:12.9284295Z   File "/var/lib/jenkins/workspace/test/functorch/test_aotdispatch.py", line 6184, in _test_aot_autograd_module_helper
2024-08-22T07:36:12.9285234Z     _test_aot_autograd_forwards_backwards_helper(
2024-08-22T07:36:12.9286567Z   File "/opt/conda/envs/py_3.12/lib/python3.12/site-packages/torch/testing/_internal/optests/aot_autograd.py", line 139, in _test_aot_autograd_forwards_backwards_helper
2024-08-22T07:36:12.9287803Z     check(args, ignore_failure=False)
2024-08-22T07:36:12.9288843Z   File "/opt/conda/envs/py_3.12/lib/python3.12/site-packages/torch/testing/_internal/optests/aot_autograd.py", line 121, in check
2024-08-22T07:36:12.9289938Z     any_non_leaves = any(x.requires_grad and not x.is_leaf for x in args)
2024-08-22T07:36:12.9290553Z                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-08-22T07:36:12.9291657Z   File "/opt/conda/envs/py_3.12/lib/python3.12/site-packages/torch/testing/_internal/optests/aot_autograd.py", line 121, in <genexpr>
2024-08-22T07:36:12.9292765Z     any_non_leaves = any(x.requires_grad and not x.is_leaf for x in args)
2024-08-22T07:36:12.9293338Z                          ^^^^^^^^^^^^^^^
2024-08-22T07:36:12.9293931Z AttributeError: 'dict' object has no attribute 'requires_grad'
2024-08-22T07:36:12.9294347Z 
2024-08-22T07:36:12.9294617Z To execute this test, run the following from the base repo dir:
2024-08-22T07:36:12.9295763Z     python test/functorch/test_aotdispatch.py TestEagerFusionModuleInfoCPU.test_aot_autograd_module_exhaustive_nn_AdaptiveAvgPool1d_cpu_float32
```

Reply:
hmm there are some new CI errors  I need to look into, but I think I fixed that one (update the PR to pytree_flatten and only look at tensor args)